polkadot/xcm/pallet-xcm-benchmarks/src/generic/mod.rs

```diff
@@ -104,6 +107,13 @@ pub mod pallet {
 				crate_version: <frame_system::Pallet<Self> as frame_support::traits::PalletInfoAccess>::crate_version(),
 			}
 		}
+
+		/// The worst case buy execution weight limit and
+		/// asset to trigger the Trader::buy_execution in the XCM executor
+		/// By default returns ((AssetId(Here.into()), 100_000_000u128), Unlimited)
+		fn worst_case_buy_execution() -> Result<(Asset, WeightLimit), BenchmarkError> {
+			Ok(((AssetId(Junctions::Here.into()), 100_000_000u128).into(), WeightLimit::Unlimited))
+		}
```

Comment:
I would not add here default, we want runtimes to set the worst case:
```suggestion
		fn worst_case_buy_execution() -> Result<(Asset, WeightLimit), BenchmarkError>;
```

Just thinking about `fn fee_asset()` which is supposed to do the same for `PayFees`,
maybe we could have something like `fn worst_case_for_trader()` and use it for `PayFees` and `BuyExecution` benchmarks.

Reply:
is there any hint that I can take for the worst case scenario of each runtime? or should I try to guess?

Reply:
actually never mind, I can get it through `fee_asset`